source/_integrations/shelly.markdown

```diff
@@ -54,5 +54,8 @@ Names are set from the device web page:
 ## Known issues and limitations
 
 - Only supports firmware 1.8 and later
+  - in case your device does not offer a firmware update, but the integration still complains about the firmware, have a look at the [Shelly Firmware Archive](http://archive.shelly-faq.de/) (which also contains the latest firmware versions)
+    1. Make sure your device is on the same network by entering its IP into the browser's address bar (the device's configuration page should appear)
+    2. Copy the IP into the *Shelly IP* field, select device and (latest) version, and follow the instructions
```

Comment:
To me, this feels a bit weird.

Can't we just link to their documentation stating how to update the firmware? The firmware is offered as an upgrade on the device itself? Why is the archive needed?

Reply:
Well, I spent like 3 hours yesterday figuring out how to update the firmware. For my old version, I would not get an upgrade offer (that's the whole point of adding that information). I then googled for quite a while, and ended up [asking on the HA forum](https://community.home-assistant.io/t/how-to-update-shelly-plug-s-firmware-to-latest-version/252060) (where I also described why I didn't find that firmware on the official webpage). I think that webpage exits for a reason ;-)

I see your concern, though (in particular because the linked page does not support https). It might be an option to add a warning... If I have overseen any obvious instructions, feel free to replace my enhancement with a link to them, but I figured I'd save other people from that waste of time I had yesterday...

Reply:
> in particular because the linked page does not support http

That is not my concern, my concern is that this feels wrong and out of place in general.

Reply:
Note that I don't want to clutter HA's pull requests or steal your time for a minor improvement - if you feel uneasy about the change, feel free to just close the PR, no worries from my side. People might now even find my HA forum entry when googling for that problem.

Reply:
As a test, I've just installed/unpacked an older shelly I had still in a box.
I could not use it with the integration (as, expected, the firmware was too old). I click on the offered upgrade button in the shelly UI and everything worked. 🤷

Reply:
As I described in my forum post linked above, I was neither offered an upgrade with the version installed on the Plug S out of the box nor with the "intermediate version" from beginning of this year. Now I am offered an upgrade to 1.9.0 (I'm on 1.9.2 now) ;-)

Note also [this forum search](https://www.shelly-support.eu/forum/index.php?board/36-shelly-firmware-update/), which lists a whole bunch of people having update problems. Why it worked for you, I don't know, but it certainly did not work for me.

One thing just came to my mind: Did you connect to the Shelly cloud for the update? Note that I did not, I did not even create a Shelly account - I used the "Access Point" method for configuration/update.

Please note that I do not want to spend the work researching why this is the case - enough Shelly googling for me for now :-) But here's one more suggestion: We could replace the change with a much shorter one in the line of 
```
If your Shelly device does not offer a firmware upgrade, have a look at [this forum post](https://community.home-assistant.io/t/how-to-update-shelly-plug-s-firmware-to-latest-version/252060).
```

Reply:
Have a look at the last (and new) comment at the [HA forum post](https://community.home-assistant.io/t/how-to-update-shelly-plug-s-firmware-to-latest-version/252060) ;-)

Reply:
This PR has been ignored for quite a while now, which is in nobody's interest, I would say. Thus, I (once more) suggest to either merge this PR (which I would obviously prefer) or to close it without merging (which I would still be fine with). To this end, I do not see how I can contribute any further (except solving the merge conflict - let me know if you want me to do this).

As a side note: I don't want to make a mountain out of a molehill, but still: The handling of this issue is not very encouraging for further contributions to the HA community.

Reply:
>  The handling of this issue is not very encouraging for further contributions to the HA community.

We have over 800 PRs open, and do what we can. If you are discouraged by the fact we think about things and try to make things work; instead of just merging what comes in. 🤷 

As for the contents of this PR, handling firmware updates is a Shelly problem and issue. I'm fine with linking to their documentation, but not with documenting third-party upgrade procedures.

Reply:
You are missing my point, which is exactly that "I don't want to clutter HA's pull requests" (as I have stated above). I don't want you to just merge things at all (I have a reasonably successful OS project myself). However, I have even suggested an alternative approach, and never got any feedback for that. That's what I'm talking about. But hey, let's stop right here - HA is a great project with a great community, no question about that.

Ok, since there is no according documentation afai know, I suggest that you close the PR.

Reply:
I've discussed this internally and pushed a signal up to our shelly contact on what is happening here.

However, we don't want to add this to our documentation for the reasons I stated earlier. If one has problems upgrading firmware on their device, one should contact the manufacturer, in this case, Shelly.